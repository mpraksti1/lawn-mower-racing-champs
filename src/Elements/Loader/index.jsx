import React from 'react'
import StyledDiv from './styles';

const Loader = () => (
  <StyledDiv>
    <div className="loader">
      <div className="slide-section-container">
        <div className="slide-section"></div>
        <div className="slide-section"></div>
        <div className="slide-section"></div>
        <div className="slide-section"></div>
        <div className="slide-section"></div>
        <div className="slide-section"></div>
        <div className="slide-section"></div>
        <div className="slide-section"></div>
      </div>
    </div>
  </StyledDiv>
);

export default Loader;