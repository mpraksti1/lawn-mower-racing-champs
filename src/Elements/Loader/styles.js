import styled, { keyframes } from 'styled-components';

const topToBottom = keyframes`
  0% {transform: translateY(50%);}
  50% {transform: translateY(0%);}
  100% {transform: translateY(-100%);}
`

const bottomToTop = keyframes`
  0% {transform: translateY(-50%);}
  50% {transform: translateY(0%);}
  100% {transform: translateY(100%);}
`

const StyledDiv = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: ${({ theme }) => theme.colors.darkGreen};
  overflow: hidden;

  .loader {
    .slide-section-container {
      width: 750px;
      transform-origin: 750px 200px;
      transform: rotate(30deg);
    }
    .slide-section {
      width: 12.5%;
      background: ${({ theme }) => theme.colors.green};
      height: 750px;
      float: left;
      &:nth-child(even) {
        transform: translateY(50%);
        animation: ${topToBottom} linear infinite alternate;
        border-radius: 10px;
      }
      &:nth-child(odd) {
        transform: translateY(-50%);
        animation: ${bottomToTop} linear infinite alternate;
        border-radius: 10px;
      }
      &:nth-child(1) {
        animation-duration: 3s;
      }
      &:nth-child(2) {
        animation-duration: 5s;
      }
      &:nth-child(3) {
        animation-duration: 7s;
      }
      &:nth-child(4) {
        animation-duration: 4s;
      }
      &:nth-child(5) {
        animation-duration: 5s;
      }
      &:nth-child(6) {
        animation-duration: 6s;
      }
      &:nth-child(7) {
        animation-duration: 10s;
      }
      &:nth-child(8) {
        animation-duration: 3s;
      }
    }
  }
`

StyledDiv.displayName = 'LoaderStyled';

export default StyledDiv;