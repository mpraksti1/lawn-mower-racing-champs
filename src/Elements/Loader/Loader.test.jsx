import Loader from '.';

describe('Loader component', () => {
  it('renders without crashing', () => {
    shallow(<Loader />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  });
});
