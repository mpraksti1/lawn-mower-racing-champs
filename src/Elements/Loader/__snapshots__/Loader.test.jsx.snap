// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component matches snapshot 1`] = `
<LoaderStyled>
  <div
    className="loader"
  >
    <div
      className="slide-section-container"
    >
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
      <div
        className="slide-section"
      >
         
      </div>
    </div>
  </div>
</LoaderStyled>
`;
